mcf-browser/src/back-end/utils.ts

```diff
@@ -169,21 +174,37 @@ function shouldReadLine(line: string) : boolean {
 }
 
 /**
- * Queries Data Commons to get the name for a location
- * @param {string} dcid The dcid to get the location for
- * @return {Promise<string>} Returns the location
+ * Queries Data Commons to get the name for a set of entities
+ * @param {string[]} entities The dcids for the entities
+ * @return {Promise<string>} Returns a list of names
  */
-async function getName(dcid: string) {
-  const url = `${API_ROOT}/v1/property/values/out/${dcid}/name`;
+async function getName(entities: string[]) {
+  if (entities.length == 0) {
+    return [];
+  }
 
-  // expected response if dcid exists is {"values":"[...]}
-  // expected response if dcid does not exist is {}
-  return fetch(url)
+  const url = `${API_ROOT}/v1/bulk/property/values/out`;
+  const data = {
+    'property': 'name',
+    'entities': entities,
+  };
+  const options = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
```

Comment:
We'll need to add an API key to this.

Reply:
Do you mean:

a. we need to add an API key in this function and pass it when making the API call - if this is the case, we want to add a TODO in the code
b. mixer needs to enforce API key for this endpoint - if this is the case, please create a github issue for this in the mixer repository